nixos/modules/system/boot/luksroot.nix

```diff
@@ -1081,6 +1096,39 @@ in
     boot.initrd.preLVMCommands = mkIf (!config.boot.initrd.systemd.enable) (commonFunctions + preCommands + concatStrings (mapAttrsToList openCommand preLVM) + postCommands);
     boot.initrd.postDeviceCommands = mkIf (!config.boot.initrd.systemd.enable) (commonFunctions + preCommands + concatStrings (mapAttrsToList openCommand postLVM) + postCommands);
 
+    boot.initrd.systemd.services = let devicesWithClevis = filterAttrs (device: _: (hasAttr device clevis.devices)) luks.devices; in
+      mkIf (clevis.enable && systemd.enable) (
+        (mapAttrs'
+          (name: _: nameValuePair "cryptsetup-clevis-${name}" {
+            wantedBy = [ "systemd-cryptsetup@${name}.service" ];
+            before = [
+              "systemd-cryptsetup@${name}.service"
+              "initrd-switch-root.target"
+              "shutdown.target"
+            ];
+            wants = [ "systemd-udev-settle.service" "network-online.target" ];
+            after = [
+              "systemd-udev-settle.service"
```

Comment:
why does this need `udev-settle`? It's best to avoid that when possible.

Reply:
Removed

Reply:
Actually, this was useful to wait for the TPM to be available. I'll put it back for now, but if you know of a better synchronization point for that I'd be interested.